lib/Email/MIME/Encode.pm

```diff
@@ -16,6 +17,10 @@ sub maybe_mime_encode_header {
     $header = lc $header;
 
     my $header_length = length($header) + length(": ");
+
+    return $val->as_mime_string($charset, $header_length)
```

Comment:
Probably `$header` and `$header_length` are misleading here. `$header` is `field name` (according to RFC2822) and `$header_length` length of `field name` + 2 (for colon and space).

Purpose of passing `$header_length` into encoder is ability to know how many octets are already print on first line before `field body`. Encoder can use this information and optimize whole header to fil into less lines. But it is just optional... Email::Simple can wrap correctly `field body` produced by encoder...

Parameter `$header_length` is optional for encoder, it does not have to use it. So I put it as second argument.

On the other hand, if `$charset` is passed, then encoder should use it and encode `field body` into that charset. Therefore I think `$charset` is more "required" as `$header_length` so I put `$charset` before `$header_length`.

If you have other opinion or better idea for naming variables just propose it...